test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
     end
   end
 
+  describe 'create_or_update_teacherset ' do
+    it 'test create or update teacherset' do
+      ts_items_info = {bibs_resp: SIERRA_USER, total_count: 1, available_count: 1, availability_string: 'available'}
+      TeacherSet.instance_variable_set(:@req_body, SIERRA_USER["data"][0])
+      bnumber = {bnumber: "b7899158"}
+      teacher_set = TeacherSet.new(bnumber: "b7899158")
+      subjects_arr = ["Elections."]
+
+      resp = nil
+      @mintest_mock2.expect(:call, ts_items_info, [7899158])
+      @mintest_mock3.expect(:call, teacher_set, [ts_items_info])
+      @mintest_mock4.expect(:call, teacher_set)
+      @mintest_mock5.expect(:call, subjects_arr, ['650'])
+      @mintest_mock6.expect(:call, teacher_set, [subjects_arr])
+      @mintest_mock7.expect(:call, "Learning set", ['500', true])
+      @mintest_mock8.expect(:call, teacher_set, ["Learning set"])
+      @mintest_mock9.expect(:call, teacher_set, [SIERRA_USER["data"][0]])
+      @mintest_mock10.expect(:call, teacher_set)
+
+      TeacherSet.stub :initialize_teacher_set, teacher_set, [7899158] do
+        teacher_set.stub :get_items_info_from_bibs_service, ts_items_info, [7899158] do
+          teacher_set.stub :update_teacher_set_attributes_from_bib_request, @mintest_mock3 do
+            teacher_set.stub :clean_primary_subject, @mintest_mock4 do
+              teacher_set.stub :all_var_fields, @mintest_mock5 do
+                teacher_set.stub :update_subjects_via_api, @mintest_mock6 do
+                  teacher_set.stub :var_field_data, @mintest_mock7 do
+                    teacher_set.stub :update_notes, @mintest_mock8 do
+                      teacher_set.stub :update_included_book_list, @mintest_mock9 do
+                        teacher_set.stub :create_or_update_teacherset_document_in_es, @mintest_mock10 do
+                          resp = TeacherSet.create_or_update_teacher_set_data(SIERRA_USER["data"][0])
+                        end
+                      end
+                    end
+                  end
+                end
+              end
+            end
+          end
+        end
+      end
+      assert_equal(teacher_set.bnumber, resp.bnumber)
+      @mintest_mock3.verify
+      @mintest_mock4.verify
+      @mintest_mock5.verify
+      @mintest_mock6.verify
+      @mintest_mock7.verify
+      @mintest_mock8.verify
+      @mintest_mock9.verify
+      @mintest_mock10.verify
+    end
+  end
+
+
+  describe 'delete teacher_set record from db and ES' do
+    it 'test delete_teacher_set_method' do
+      bib_id = '1234'
+      teacher_set = TeacherSet.new(bnumber: "b#{bib_id}")
+      @mintest_mock1.expect(:call, teacher_set, [bib_id])
+  
+
+      resp = nil
+      TeacherSet.stub :get_teacher_set_by_bnumber, @mintest_mock1 do
+        teacher_set.stub :destroy, teacher_set do
+          teacher_set.stub :delete_teacherset_record_from_es, teacher_set do
```

Comment:
So for this one, I think `delete_teacherset_record_from_es` should be mocked, but I don't think anything else need be.